gcloud/orb.yaml

```diff
@@ -122,7 +122,7 @@ commands:
           name: install dependencies
           command: |
             if which apk >/dev/null; then
-              apk add --no-cache --no-progress bash curl python3
+              apk add --no-cache --no-progress bash curl curl-dev python3
```

Comment:
Is this always guaranteed to fix the issue? It feels a bit like "programming by coincidence", unless this is the recommended solution. I wonder why this happens in the first place. I don't seem to get this issue on `alpine:latest`. On which version of `alpine` did this occur? Perhaps we should consider it borked and avoid it. Or -- perhaps the problem is `libcurl` was already installed whenever this gcloud orb command was run to install `curl`, and we never needed to actually install `curl` in the first place?

I guess I should just ask: in what specific context did this error occur? :P

Reply:
* this occurred on the `image: docker:20.10.21-git` alpine
* I checked the already installed idea by just starting an empty orb, checking that curl was not installed, installing it only, and then fixing the issue with curl-dev

Reply:
Perhaps we should be running `apk update` before the `curl` install similar to how we do below for other distributions. That also seems to fix the problem for this particular image.

Reply:
Oh, interesting, if an `apk` update solves it that'd make sense to me. I was under the impression `apk add` with `--no-cache` called an `update` anyway (since it can't use the cache, so has to pull new data), but that could entirely have been a misread on my part.

Reply:
Oh shoot, `--no-cache` is the version we wanted for docker images (well, in the past: now that cache-mounts at build time are a thing that's not true either). I think we should be using `--update-cache` instead...

Reply:
hm `--update-cache` didnt solve the problem, going to change to running an `apk update` I think thats the cleanest

Reply:
oh interesting apk update didnt work either...